src/applications/bmqstoragetool/m_bmqstoragetool_cslprinter.cpp

```diff
@@ -0,0 +1,568 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <m_bmqstoragetool_cslprinter.h>
+
+// bmqstoragetool
+#include <m_bmqstoragetool_parameters.h>
+
+// BMQ
+#include <bmqu_alignedprinter.h>
+#include <bmqu_jsonprinter.h>
+#include <bmqu_memoutstream.h>
+
+// BDE
+#include <bslim_printer.h>
+
+namespace BloombergLP {
+namespace m_bmqstoragetool {
+
+namespace {
+
+// Helper to convert record to json string (escaped single line)
+template <typename RECORD_TYPE>
+bsl::string recordToJsonString(const RECORD_TYPE* record,
+                               bslma::Allocator*  allocator,
+                               bool               removeTrailingQuotes = true)
+{
+    bmqu::MemOutStream recStr(allocator);
+    // Print record in one line string
+    record->print(recStr, -1, -1);
+
+    // Escape the string
+    bmqu::MemOutStream escapedStr(allocator);
+    escapedStr << bsl::quoted(recStr.str());
+
+    bsl::string resultStr = escapedStr.str();
+    // Remove leading and trailing quotes with '\n' character
+    if (removeTrailingQuotes && !resultStr.empty() &&
+        resultStr.front() == '"' && resultStr.back() == '"') {
+        resultStr = resultStr.substr(1, resultStr.size() - 3);
+    }
+
+    return resultStr;
+}
+
+// Helper to print queue info in json format
+void printQueueInfo(bsl::ostream&     ostream,
+                    const QueueMap&   queueMap,
+                    unsigned int      queuesLimit,
+                    bslma::Allocator* allocator)
+{
+    const QueueInfos& queueInfos = queueMap.queueInfos();
+    if (!queueInfos.empty()) {
+        ostream << ",\n";
+        QueueInfos::const_iterator itEnd = queueInfos.cend();
+        if (queueInfos.size() > queuesLimit) {
+            ostream << "    \"First" << queuesLimit << "Queues\": [";
+            itEnd = queueInfos.cbegin() + queuesLimit;
+        }
+        else {
+            ostream << "    \"Queues\": [";
+        }
+        QueueInfos::const_iterator it = queueInfos.cbegin();
+        for (; it != itEnd; ++it) {
+            if (it != queueInfos.cbegin()) {
+                ostream << ",";
+            }
+            bsl::string recStr = recordToJsonString(it, allocator, false);
+
+            ostream << "\n      " << recStr;
+        }
+        ostream << "\n    ]";
+    }
+    else {
+        ostream << "    \"Queues\": []";
+    }
+}
+
+}  // close unnamed namespace
+
+// =============================
+// class HumanReadableCslPrinter
+// =============================
+
+class HumanReadableCslPrinter : public CslPrinter {
+  private:
+    // PRIVATE DATA
+
+    bsl::ostream&     d_ostream;
+    bslma::Allocator* d_allocator_p;
+
+  public:
+    // CREATORS
+    HumanReadableCslPrinter(bsl::ostream& os, bslma::Allocator* allocator)
```

Comment:
```suggestion
    explicit HumanReadableCslPrinter(bsl::ostream& os, bslma::Allocator* allocator = 0)
```
+ nested trait uses bslma allocator